### ISSUE TITLE
Build failed on kernel 6.8.1

### ISSUE THREAD

**Issue by olm52** [bondagit/ravenna-alsa-lkm#17]
```
DKMS make.log for ravenna-alsa-lkm-r141.6ca0a3d for kernel 6.8.1-arch1-1.1 (x86_64)
lun 18 mar 2024, 10:21:59, CET
make -C /lib/modules/6.8.1-arch1-1.1/build/ M=/var/lib/dkms/ravenna-alsa-lkm/r141.6ca0a3d/build modules
make[1]: Entering directory '/usr/lib/modules/6.8.1-arch1-1.1/build'
  CC [M]  /var/lib/dkms/ravenna-alsa-lkm/r141.6ca0a3d/build/c_wrapper_lib.o
  CC [M]  /var/lib/dkms/ravenna-alsa-lkm/r141.6ca0a3d/build/audio_driver.o
  CC [M]  /var/lib/dkms/ravenna-alsa-lkm/r141.6ca0a3d/build/manager.o
  CC [M]  /var/lib/dkms/ravenna-alsa-lkm/r141.6ca0a3d/build/module_main.o
  CC [M]  /var/lib/dkms/ravenna-alsa-lkm/r141.6ca0a3d/build/PTP.o
  CC [M]  /var/lib/dkms/ravenna-alsa-lkm/r141.6ca0a3d/build/module_interface.o
  CC [M]  /var/lib/dkms/ravenna-alsa-lkm/r141.6ca0a3d/build/module_netlink.o
  CC [M]  /var/lib/dkms/ravenna-alsa-lkm/r141.6ca0a3d/build/module_timer.o
  CC [M]  /var/lib/dkms/ravenna-alsa-lkm/r141.6ca0a3d/build/EtherTubeNetfilter.o
  CC [M]  /var/lib/dkms/ravenna-alsa-lkm/r141.6ca0a3d/build/RTP_streams_manager.o
  CC [M]  /var/lib/dkms/ravenna-alsa-lkm/r141.6ca0a3d/build/RTP_audio_stream.o
  CC [M]  /var/lib/dkms/ravenna-alsa-lkm/r141.6ca0a3d/build/RTP_stream.o
  CC [M]  /var/lib/dkms/ravenna-alsa-lkm/r141.6ca0a3d/build/RTP_stream_info.o
  CC [M]  /var/lib/dkms/ravenna-alsa-lkm/r141.6ca0a3d/build/MTAL_EthUtils.o
  CC [M]  /var/lib/dkms/ravenna-alsa-lkm/r141.6ca0a3d/build/MTAL_LKernelAPI.o
  CC [M]  /var/lib/dkms/ravenna-alsa-lkm/r141.6ca0a3d/build/MTConvert.o
/var/lib/dkms/ravenna-alsa-lkm/r141.6ca0a3d/build/manager.c:1319:5: warning: no previous prototype for ‘pdate_live_in_audio_data_format’ [-Wmissing-prototypes]
 1319 | int pdate_live_in_audio_data_format(void* user, uint32_t ulChannelId, char const * pszCodec)
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/var/lib/dkms/ravenna-alsa-lkm/r141.6ca0a3d/build/module_timer.c:69:6: warning: no previous prototype for ‘tasklet_hrtimer_init’ [-Wmissing-prototypes]
   69 | void tasklet_hrtimer_init(struct tasklet_hrtimer *ttimer,
      |      ^~~~~~~~~~~~~~~~~~~~
/var/lib/dkms/ravenna-alsa-lkm/r141.6ca0a3d/build/module_timer.c:94:22: warning: no previous prototype for ‘timer_callback’ [-Wmissing-prototypes]
   94 | enum hrtimer_restart timer_callback(struct hrtimer *timer)
      |                      ^~~~~~~~~~~~~~
/var/lib/dkms/ravenna-alsa-lkm/r141.6ca0a3d/build/PTP.c:77:10: warning: no previous prototype for ‘get_FS’ [-Wmissing-prototypes]
   77 | uint32_t get_FS(uint32_t ui32SamplingRate)
      |          ^~~~~~
/var/lib/dkms/ravenna-alsa-lkm/r141.6ca0a3d/build/PTP.c:101:10: warning: no previous prototype for ‘get_samplerate_base’ [-Wmissing-prototypes]
  101 | uint32_t get_samplerate_base(uint32_t ui32SamplingRate)
      |          ^~~~~~~~~~~~~~~~~~~
/var/lib/dkms/ravenna-alsa-lkm/r141.6ca0a3d/build/PTP.c:138:17: warning: no previous prototype for ‘SetSeconds’ [-Wmissing-prototypes]
  138 | /*static*/ void SetSeconds(uint64_t ui64Time, uint8_t bySeconds[6])
      |                 ^~~~~~~~~~
/var/lib/dkms/ravenna-alsa-lkm/r141.6ca0a3d/build/PTP.c:148:17: warning: no previous prototype for ‘DumpPTPV2MsgHeader’ [-Wmissing-prototypes]
  148 | /*static*/ void DumpPTPV2MsgHeader(TV2MsgHeader* pV2MsgHeader)
      |                 ^~~~~~~~~~~~~~~~~~
/var/lib/dkms/ravenna-alsa-lkm/r141.6ca0a3d/build/PTP.c:164:17: warning: no previous prototype for ‘DumpV2TimeRepresentation’ [-Wmissing-prototypes]
  164 | /*static*/ void DumpV2TimeRepresentation(TV2TimeRepresentation* pOriginTimestamp)
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~
/var/lib/dkms/ravenna-alsa-lkm/r141.6ca0a3d/build/audio_driver.c: In function ‘mr_alsa_audio_chip_probe’:
/var/lib/dkms/ravenna-alsa-lkm/r141.6ca0a3d/build/audio_driver.c:2170:5: error: implicit declaration of function ‘strlcpy’; did you mean ‘strscpy’? [-Werror=implicit-function-declaration]
 2170 |     strlcpy(card->driver, SND_MR_ALSA_AUDIO_DRIVER, sizeof(card->driver));
      |     ^~~~~~~
      |     strscpy
/var/lib/dkms/ravenna-alsa-lkm/r141.6ca0a3d/build/MTAL_LKernelAPI.c:67:6: warning: no previous prototype for ‘__cxa_pure_virtual’ [-Wmissing-prototypes]
   67 | void __cxa_pure_virtual(void)
      |      ^~~~~~~~~~~~~~~~~~
/var/lib/dkms/ravenna-alsa-lkm/r141.6ca0a3d/build/module_interface.c:71:14: warning: no previous prototype for ‘nf_hook_func’ [-Wmissing-prototypes]
   71 | unsigned int nf_hook_func(unsigned int hooknum, struct sk_buff *skb, const struct net_device *in, const struct net_device *out, int (*okfn)(struct sk_buff *))
      |              ^~~~~~~~~~~~
/var/lib/dkms/ravenna-alsa-lkm/r141.6ca0a3d/build/module_interface.c:192:10: warning: no previous prototype for ‘__aeabi_uldivmod’ [-Wmissing-prototypes]
  192 | uint64_t __aeabi_uldivmod(uint64_t numerator, uint64_t denominator)
      |          ^~~~~~~~~~~~~~~~
/var/lib/dkms/ravenna-alsa-lkm/r141.6ca0a3d/build/module_interface.c:197:9: warning: no previous prototype for ‘__aeabi_ldivmod’ [-Wmissing-prototypes]
  197 | int64_t __aeabi_ldivmod(int64_t numerator, int64_t denominator)
      |         ^~~~~~~~~~~~~~~
/var/lib/dkms/ravenna-alsa-lkm/r141.6ca0a3d/build/module_interface.c:203:10: warning: no previous prototype for ‘__udivdi3’ [-Wmissing-prototypes]
  203 | uint64_t __udivdi3(uint64_t numerator, uint64_t denominator)
      |          ^~~~~~~~~
/var/lib/dkms/ravenna-alsa-lkm/r141.6ca0a3d/build/module_interface.c:209:9: warning: no previous prototype for ‘__divdi3’ [-Wmissing-prototypes]
  209 | int64_t __divdi3(int64_t numerator, int64_t denominator)
      |         ^~~~~~~~
/var/lib/dkms/ravenna-alsa-lkm/r141.6ca0a3d/build/module_interface.c:215:10: warning: no previous prototype for ‘__umoddi3’ [-Wmissing-prototypes]
  215 | uint64_t __umoddi3(uint64_t numerator, uint64_t denominator)
      |          ^~~~~~~~~
cc1: some warnings being treated as errors
make[3]: *** [scripts/Makefile.build:243: /var/lib/dkms/ravenna-alsa-lkm/r141.6ca0a3d/build/audio_driver.o] Error 1
make[2]: *** [/usr/lib/modules/6.8.1-arch1-1.1/build/Makefile:1921: /var/lib/dkms/ravenna-alsa-lkm/r141.6ca0a3d/build] Error 2
make[1]: *** [Makefile:240: __sub-make] Error 2
make[1]: Leaving directory '/usr/lib/modules/6.8.1-arch1-1.1/build'
make: *** [Makefile:15: modules] Error 2

```

**Comment by bondagit** [comment]
This should be fixed with the driver version v1.8, can you pull the changes and re-try ?

**Comment by olm52** [comment]
Just tested. It is compiling fine, thanks.
